lib/ansible/modules/cloud/docker/docker_compose.py

```diff
@@ -866,9 +866,12 @@ def cmd_pull(self):
                 except Exception as exc:
                     self.client.fail("Error: service image lookup failed - %s" % str(exc))
 
+                out_redir_name, err_redir_name = make_redirection_tempfiles()
                 # pull the image
                 try:
-                    service.pull(ignore_pull_failures=False)
+                    with stdout_redirector(out_redir_name):
+                        with stderr_redirector(err_redir_name):
+                            service.pull(ignore_pull_failures=False)
                 except Exception as exc:
                     self.client.fail("Error: pull failed with %s" % str(exc))
```

Comment:
The other places where `make_redirection_tempfiles()` is used are doing the following:
```suggestion
                    fail_reason = get_failure_info(exc, out_redir_name, err_redir_name,
                                                   msg_format="Error: pull failed with %s")
                    self.client.fail(**fail_reason)
                else:
                    cleanup_redirection_tempfiles(out_redir_name, err_redir_name)
```
That also cleans up the output files in both error and non-error case.

(Same for the other places.)

Reply:
Aha! I was actually wondering about this. I'll get that added as well!